tests/integration/security/reachability_test.go

```diff
@@ -119,22 +127,38 @@ func TestReachability(t *testing.T) {
 					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
 						// autoMtls doesn't work for client that doesn't have proxy, unless target doesn't
 						// have proxy or have mTLS disabled
-						if src == rctx.Naked {
-							return opts.Target == rctx.Naked || (opts.Target == rctx.B && opts.PortName != "http")
+						if rctx.IsNaked(src) {
+							return rctx.IsNaked(opts.Target) || (opts.Target == rctx.B && opts.PortName != "http")
 
 						}
+						// Headless service with sidecar injected, autoMtls does not apply.
+						if src != rctx.Headless && opts.Target == rctx.Headless {
+							return false
+						}
 						// PeerAuthentication disable mTLS for workload app:b, except http port. Thus, autoMTLS
 						// will fail on all ports on b, except http port.
 						return opts.Target != rctx.B || opts.PortName == "http"
 					},
 				},
+				{
+					// test access headless service with sidecar injected
+					ConfigFile:          "beta-mtls-off-headless.yaml",
```

Comment:
this comment is inconsistent with the comment you put the yaml file.

can you say

```
test access headless service with mTLS off to simulate without sidecar.
```